lib/queue.js

```diff
@@ -1,33 +1,52 @@
 class Queue {
     constructor() {
-        // this.store = ...
-        throw new Error("This method has not been implemented!");
+        this.store = new Array(this.capacity);
+        this.length = 0;
+        this.capacity = 20;
+        this.head = 0;
+        this.tail = -1;
     }
 
     enqueue(element) {
-        throw new Error("This method has not been implemented!");
+        if (this.length >= this.capacity) return;
+
+        this.tail = (this.tail + 1) % this.capacity;
+        this.store[this.tail % this.capacity] = element;
+        this.length++;
     }
 
     dequeue() {
-        throw new Error("This method has not been implemented!");
+        if (this.isEmpty()) return;
+
+        let result = this.store[this.head % this.capacity];
+
+        this.store[this.head % this.capacity] = null;
+
+        this.head = (this.head + 1) % this.capacity;
+        this.length--;
+        return result;
     }
 
     front() {
-        throw new Error("This method has not been implemented!");
+        return this.head;
```

Comment:
Here we want to return the value stored at `this.head` not the value stored in the pointer itself